[INFO] [2019-07-27 18:58:44][io.netty.handler.logging.LoggingHandler][id: 0xa9449dd1] REGISTERED
[INFO] [2019-07-27 18:58:44][io.netty.handler.logging.LoggingHandler][id: 0xa9449dd1] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 18:58:44][io.netty.handler.logging.LoggingHandler][id: 0xa9449dd1, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 18:59:56][io.netty.handler.logging.LoggingHandler][id: 0xa9449dd1, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xdd2cdf9b, L:/127.0.0.1:12302 - R:/127.0.0.1:19502]
[INFO] [2019-07-27 19:05:23][io.netty.handler.logging.LoggingHandler][id: 0xb776c7ba] REGISTERED
[INFO] [2019-07-27 19:05:23][io.netty.handler.logging.LoggingHandler][id: 0xb776c7ba] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:05:23][io.netty.handler.logging.LoggingHandler][id: 0xb776c7ba, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:05:31][io.netty.handler.logging.LoggingHandler][id: 0xb776c7ba, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xa2ef209a, L:/127.0.0.1:12302 - R:/127.0.0.1:19651]
[WARN] [2019-07-27 19:08:20][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:366)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
[INFO] [2019-07-27 19:08:30][io.netty.handler.logging.LoggingHandler][id: 0x62caea8f] REGISTERED
[INFO] [2019-07-27 19:08:30][io.netty.handler.logging.LoggingHandler][id: 0x62caea8f] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:08:30][io.netty.handler.logging.LoggingHandler][id: 0x62caea8f, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:08:36][io.netty.handler.logging.LoggingHandler][id: 0x62caea8f, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xa6318e2c, L:/127.0.0.1:12302 - R:/127.0.0.1:19783]
[WARN] [2019-07-27 19:08:53][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:366)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
[INFO] [2019-07-27 19:11:00][io.netty.handler.logging.LoggingHandler][id: 0x961e1ba8] REGISTERED
[INFO] [2019-07-27 19:11:00][io.netty.handler.logging.LoggingHandler][id: 0x961e1ba8] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:11:00][io.netty.handler.logging.LoggingHandler][id: 0x961e1ba8, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:11:05][io.netty.handler.logging.LoggingHandler][id: 0x961e1ba8, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xf5f05671, L:/127.0.0.1:12302 - R:/127.0.0.1:19877]
[WARN] [2019-07-27 19:11:11][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:366)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
[INFO] [2019-07-27 19:27:20][io.netty.handler.logging.LoggingHandler][id: 0xfab5af00] REGISTERED
[INFO] [2019-07-27 19:27:20][io.netty.handler.logging.LoggingHandler][id: 0xfab5af00] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:27:20][io.netty.handler.logging.LoggingHandler][id: 0xfab5af00, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:27:30][io.netty.handler.logging.LoggingHandler][id: 0xfab5af00, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0x26f706cc, L:/127.0.0.1:12302 - R:/127.0.0.1:20131]
[INFO] [2019-07-27 19:30:36][io.netty.handler.logging.LoggingHandler][id: 0x7f515740] REGISTERED
[INFO] [2019-07-27 19:30:36][io.netty.handler.logging.LoggingHandler][id: 0x7f515740] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:30:36][io.netty.handler.logging.LoggingHandler][id: 0x7f515740, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:30:41][io.netty.handler.logging.LoggingHandler][id: 0x7f515740, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0x25044bd2, L:/127.0.0.1:12302 - R:/127.0.0.1:20218]
[INFO] [2019-07-27 19:33:10][io.netty.handler.logging.LoggingHandler][id: 0xb6f298f0] REGISTERED
[INFO] [2019-07-27 19:33:10][io.netty.handler.logging.LoggingHandler][id: 0xb6f298f0] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:33:10][io.netty.handler.logging.LoggingHandler][id: 0xb6f298f0, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:33:14][io.netty.handler.logging.LoggingHandler][id: 0xb6f298f0, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0x7bcef6ca, L:/127.0.0.1:12302 - R:/127.0.0.1:20306]
[INFO] [2019-07-27 19:44:18][io.netty.handler.logging.LoggingHandler][id: 0xb9336d51] REGISTERED
[INFO] [2019-07-27 19:44:18][io.netty.handler.logging.LoggingHandler][id: 0xb9336d51] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:44:18][io.netty.handler.logging.LoggingHandler][id: 0xb9336d51, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:44:24][io.netty.handler.logging.LoggingHandler][id: 0xb9336d51, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0x186859ef, L:/127.0.0.1:12302 - R:/127.0.0.1:20528]
[INFO] [2019-07-27 19:46:09][io.netty.handler.logging.LoggingHandler][id: 0x0c294b3c] REGISTERED
[INFO] [2019-07-27 19:46:09][io.netty.handler.logging.LoggingHandler][id: 0x0c294b3c] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:46:09][io.netty.handler.logging.LoggingHandler][id: 0x0c294b3c, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:46:19][io.netty.handler.logging.LoggingHandler][id: 0x0c294b3c, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xbb13bcb5, L:/127.0.0.1:12302 - R:/127.0.0.1:20607]
[INFO] [2019-07-27 19:54:45][io.netty.handler.logging.LoggingHandler][id: 0xef71081f] REGISTERED
[INFO] [2019-07-27 19:54:45][io.netty.handler.logging.LoggingHandler][id: 0xef71081f] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:54:45][io.netty.handler.logging.LoggingHandler][id: 0xef71081f, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:54:52][io.netty.handler.logging.LoggingHandler][id: 0xef71081f, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0x8ac0a82a, L:/127.0.0.1:12302 - R:/127.0.0.1:20780]
[INFO] [2019-07-27 19:55:32][io.netty.handler.logging.LoggingHandler][id: 0x357441de] REGISTERED
[INFO] [2019-07-27 19:55:32][io.netty.handler.logging.LoggingHandler][id: 0x357441de] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:55:32][io.netty.handler.logging.LoggingHandler][id: 0x357441de, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:55:39][io.netty.handler.logging.LoggingHandler][id: 0x357441de, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xfa01f19a, L:/127.0.0.1:12302 - R:/127.0.0.1:20848]
[INFO] [2019-07-27 19:56:25][io.netty.handler.logging.LoggingHandler][id: 0x7c44144c] REGISTERED
[INFO] [2019-07-27 19:56:25][io.netty.handler.logging.LoggingHandler][id: 0x7c44144c] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:56:25][io.netty.handler.logging.LoggingHandler][id: 0x7c44144c, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:56:29][io.netty.handler.logging.LoggingHandler][id: 0x7c44144c, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xca68a909, L:/127.0.0.1:12302 - R:/127.0.0.1:20918]
[INFO] [2019-07-27 19:58:06][io.netty.handler.logging.LoggingHandler][id: 0xe3b2100b] REGISTERED
[INFO] [2019-07-27 19:58:06][io.netty.handler.logging.LoggingHandler][id: 0xe3b2100b] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:58:06][io.netty.handler.logging.LoggingHandler][id: 0xe3b2100b, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:58:09][io.netty.handler.logging.LoggingHandler][id: 0xe3b2100b, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0x870c9b3e, L:/127.0.0.1:12302 - R:/127.0.0.1:20997]
[INFO] [2019-07-27 19:59:21][io.netty.handler.logging.LoggingHandler][id: 0xe26c6619] REGISTERED
[INFO] [2019-07-27 19:59:21][io.netty.handler.logging.LoggingHandler][id: 0xe26c6619] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 19:59:21][io.netty.handler.logging.LoggingHandler][id: 0xe26c6619, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 19:59:26][io.netty.handler.logging.LoggingHandler][id: 0xe26c6619, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xd302c2e8, L:/127.0.0.1:12302 - R:/127.0.0.1:21074]
[INFO] [2019-07-27 20:00:03][io.netty.handler.logging.LoggingHandler][id: 0xabd2457b] REGISTERED
[INFO] [2019-07-27 20:00:03][io.netty.handler.logging.LoggingHandler][id: 0xabd2457b] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 20:00:03][io.netty.handler.logging.LoggingHandler][id: 0xabd2457b, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 20:00:09][io.netty.handler.logging.LoggingHandler][id: 0xabd2457b, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0x78537773, L:/127.0.0.1:12302 - R:/127.0.0.1:21143]
[INFO] [2019-07-27 20:06:13][io.netty.handler.logging.LoggingHandler][id: 0x160e2a74] REGISTERED
[INFO] [2019-07-27 20:06:13][io.netty.handler.logging.LoggingHandler][id: 0x160e2a74] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 20:06:13][io.netty.handler.logging.LoggingHandler][id: 0x160e2a74, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 20:06:19][io.netty.handler.logging.LoggingHandler][id: 0x160e2a74, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xbf505b3c, L:/127.0.0.1:12302 - R:/127.0.0.1:21357]
[INFO] [2019-07-27 20:08:44][io.netty.handler.logging.LoggingHandler][id: 0x9627e94b] REGISTERED
[INFO] [2019-07-27 20:08:44][io.netty.handler.logging.LoggingHandler][id: 0x9627e94b] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 20:08:44][io.netty.handler.logging.LoggingHandler][id: 0x9627e94b, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 20:08:51][io.netty.handler.logging.LoggingHandler][id: 0x9627e94b, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0x5718f65f, L:/127.0.0.1:12302 - R:/127.0.0.1:21472]
[INFO] [2019-07-27 20:13:41][io.netty.handler.logging.LoggingHandler][id: 0x4029b86d] REGISTERED
[INFO] [2019-07-27 20:13:41][io.netty.handler.logging.LoggingHandler][id: 0x4029b86d] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 20:13:41][io.netty.handler.logging.LoggingHandler][id: 0x4029b86d, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 20:13:46][io.netty.handler.logging.LoggingHandler][id: 0x4029b86d, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xf48f9015, L:/127.0.0.1:12302 - R:/127.0.0.1:21576]
[INFO] [2019-07-27 20:17:03][io.netty.handler.logging.LoggingHandler][id: 0xc7aec946] REGISTERED
[INFO] [2019-07-27 20:17:03][io.netty.handler.logging.LoggingHandler][id: 0xc7aec946] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 20:17:03][io.netty.handler.logging.LoggingHandler][id: 0xc7aec946, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 20:17:06][io.netty.handler.logging.LoggingHandler][id: 0xc7aec946, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xbaf2b8a5, L:/127.0.0.1:12302 - R:/127.0.0.1:21665]
[INFO] [2019-07-27 20:17:47][io.netty.handler.logging.LoggingHandler][id: 0x0a353669] REGISTERED
[INFO] [2019-07-27 20:17:47][io.netty.handler.logging.LoggingHandler][id: 0x0a353669] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 20:17:47][io.netty.handler.logging.LoggingHandler][id: 0x0a353669, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 20:17:53][io.netty.handler.logging.LoggingHandler][id: 0x0a353669, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xb681afcd, L:/127.0.0.1:12302 - R:/127.0.0.1:21744]
[INFO] [2019-07-27 20:18:48][io.netty.handler.logging.LoggingHandler][id: 0xe3ab41b8] REGISTERED
[INFO] [2019-07-27 20:18:48][io.netty.handler.logging.LoggingHandler][id: 0xe3ab41b8] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 20:18:48][io.netty.handler.logging.LoggingHandler][id: 0xe3ab41b8, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 20:18:59][io.netty.handler.logging.LoggingHandler][id: 0xe3ab41b8, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0x10856275, L:/127.0.0.1:12302 - R:/127.0.0.1:21819]
[WARN] [2019-07-27 20:18:59][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.TooLongFrameException: frame length (over 21) exceeds the allowed maximum (20)
	at io.netty.handler.codec.LineBasedFrameDecoder.fail(LineBasedFrameDecoder.java:146)
	at io.netty.handler.codec.LineBasedFrameDecoder.decode(LineBasedFrameDecoder.java:118)
	at io.netty.handler.codec.DelimiterBasedFrameDecoder.decode(DelimiterBasedFrameDecoder.java:232)
	at io.netty.handler.codec.DelimiterBasedFrameDecoder.decode(DelimiterBasedFrameDecoder.java:216)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:411)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:248)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:266)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:129)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
[INFO] [2019-07-27 20:19:31][io.netty.handler.logging.LoggingHandler][id: 0x6cd04315] REGISTERED
[INFO] [2019-07-27 20:19:31][io.netty.handler.logging.LoggingHandler][id: 0x6cd04315] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 20:19:31][io.netty.handler.logging.LoggingHandler][id: 0x6cd04315, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 20:19:39][io.netty.handler.logging.LoggingHandler][id: 0x6cd04315, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xdd89d40a, L:/127.0.0.1:12302 - R:/127.0.0.1:21893]
[WARN] [2019-07-27 20:19:40][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.TooLongFrameException: frame length (over 32) exceeds the allowed maximum (20)
	at io.netty.handler.codec.LineBasedFrameDecoder.fail(LineBasedFrameDecoder.java:146)
	at io.netty.handler.codec.LineBasedFrameDecoder.decode(LineBasedFrameDecoder.java:118)
	at io.netty.handler.codec.DelimiterBasedFrameDecoder.decode(DelimiterBasedFrameDecoder.java:232)
	at io.netty.handler.codec.DelimiterBasedFrameDecoder.decode(DelimiterBasedFrameDecoder.java:216)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:411)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:248)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:266)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:129)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
[INFO] [2019-07-27 20:20:57][io.netty.handler.logging.LoggingHandler][id: 0xa735a5a7] REGISTERED
[INFO] [2019-07-27 20:20:57][io.netty.handler.logging.LoggingHandler][id: 0xa735a5a7] BIND: 0.0.0.0/0.0.0.0:12302
[INFO] [2019-07-27 20:20:57][io.netty.handler.logging.LoggingHandler][id: 0xa735a5a7, L:/0:0:0:0:0:0:0:0:12302] ACTIVE
[INFO] [2019-07-27 20:21:03][io.netty.handler.logging.LoggingHandler][id: 0xa735a5a7, L:/0:0:0:0:0:0:0:0:12302] RECEIVED: [id: 0xfc2d7815, L:/127.0.0.1:12302 - R:/127.0.0.1:21985]
[WARN] [2019-08-02 09:16:32][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:366)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
[WARN] [2019-08-02 09:17:53][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:366)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
[WARN] [2019-08-02 09:18:47][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:366)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
[WARN] [2019-08-02 13:44:17][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:366)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
[WARN] [2019-08-02 13:51:09][io.netty.channel.DefaultChannelPipeline]An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:366)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
